presto-main/src/test/java/io/prestosql/sql/planner/TestLocalDynamicFiltersCollector.java

```diff
@@ -29,25 +33,64 @@
-        assertEquals(collector.getPredicate(), TupleDomain.all());
+        assertEquals(collector.getDynamicFilter(probeSymbols), TupleDomain.all());
 
-        collector.intersect(TupleDomain.all());
-        assertEquals(collector.getPredicate(), TupleDomain.all());
+        collector.addDynamicFilter(ImmutableMap.of());
+        assertEquals(collector.getDynamicFilter(probeSymbols), TupleDomain.all());
 
-        collector.intersect(tupleDomain(symbol, 1L, 2L));
-        assertEquals(collector.getPredicate(), tupleDomain(symbol, 1L, 2L));
+        collector.addDynamicFilter(toDomainMap(symbol, 1L, 2L));
+        assertEquals(collector.getDynamicFilter(probeSymbols), tupleDomain(symbol, 1L, 2L));
 
-        collector.intersect(tupleDomain(symbol, 2L, 3L));
-        assertEquals(collector.getPredicate(), tupleDomain(symbol, 2L));
+        collector.addDynamicFilter(toDomainMap(symbol, 2L, 3L));
+        assertEquals(collector.getDynamicFilter(probeSymbols), tupleDomain(symbol, 2L));
 
-        collector.intersect(tupleDomain(symbol, 0L));
-        assertEquals(collector.getPredicate(), TupleDomain.none());
+        collector.addDynamicFilter(toDomainMap(symbol, 0L));
+        assertEquals(collector.getDynamicFilter(probeSymbols), TupleDomain.none());
+    }
+
+    @Test
+    public void testCollectorMultipleScans()
+    {
+        Symbol symbol1 = new Symbol("symbol1");
+        Symbol symbol2 = new Symbol("symbol2");
+        Set<Symbol> probeSymbols1 = ImmutableSet.of(symbol1);
+        Set<Symbol> probeSymbols2 = ImmutableSet.of(symbol2);
+
+        LocalDynamicFiltersCollector collector = new LocalDynamicFiltersCollector();
+        assertEquals(collector.getDynamicFilter(probeSymbols1), TupleDomain.all());
+        assertEquals(collector.getDynamicFilter(probeSymbols2), TupleDomain.all());
+
+        collector.addDynamicFilter(toDomainMap(symbol1, 1L, 2L));
+        collector.addDynamicFilter(toDomainMap(symbol2, 2L, 3L));
+        assertEquals(collector.getDynamicFilter(probeSymbols1), tupleDomain(symbol1, 1L, 2L));
+        assertEquals(collector.getDynamicFilter(probeSymbols2), tupleDomain(symbol2, 2L, 3L));
+        assertEquals(collector.getDynamicFilter(
+                ImmutableSet.of(symbol1, symbol2)),
+                TupleDomain.withColumnDomains(ImmutableMap.of(
+                        symbol1, Domain.multipleValues(BIGINT, ImmutableList.of(1L, 2L)),
+                        symbol2, Domain.multipleValues(BIGINT, ImmutableList.of(2L, 3L)))));
+
+        collector.addDynamicFilter(toDomainMap(symbol1, 0L));
+        assertEquals(collector.getDynamicFilter(probeSymbols1), TupleDomain.none());
+        assertEquals(collector.getDynamicFilter(probeSymbols2), tupleDomain(symbol2, 2L, 3L));
+        assertEquals(collector.getDynamicFilter(ImmutableSet.of(symbol1, symbol2)), TupleDomain.none());
+
+        collector.addDynamicFilter(ImmutableMap.of(symbol2, Domain.none(BIGINT)));
```

Comment:
Please also make sure that adding `none` domain for particular symbol doesn't affect other symbols (set them to none domain too)

Reply:
Added another test case `TestLocalDynamicFiltersCollector#testCollectorMultipleScansNone`